src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -950,7 +950,7 @@ private void ResetConnection()
                 // distributed transaction - otherwise don't reset!
                 // Prepare the parser for the connection reset - the next time a trip
                 // to the server is made.
-                _parser.PrepareResetConnection(IsTransactionRoot && !IsNonPoolableTransactionRoot);
+                _parser.PrepareResetConnection(EnlistedTransaction != null && Pool != null);
```

Comment:
I don't particularly care, but is this a scenario for using the fancy pants new `is not null` ?